### DISCUSSION TITLE
Streamlit framework consideration into F´ GDS

### DISCUSSION THREAD

**Discussion by ThibFrgsGmz** [nasa/fprime#1121] [Q&A] [answered]
I have a small question about the technologies used by F´ GDS .

I am not an expert in web technologies, but from what I understand, this WebApp uses two frameworks: the Flask framework in Python and the Vue.js farmework in Javascript.

Did you consider creating this WebApp entirely in Python using the [Streamlit framework](https://streamlit.io/) at the time?

We are evaluating this framework on other projects, and the first feedback is very positive. It would allow system engineers, without advanced programming skills, to easily customize their dashboard to meet their needs. It would be higher level than Flask and would allow to include data visualization widgets very easily.

I have a hunch that it would be useful to implement it on the F´ GDS.

To whet your appetite, [here](https://www.youtube.com/watch?v=0E_31WqVzCY) is a quick tutorial on how to implement data visualization widgets.

Streamlit GitHub: https://github.com/streamlit/streamlit

**Comment by LeStarch** [comment, marked as answer]
We did not evaluate this framework.  I can give a bit of insight into the evaluation process:

1. We've had lots of issues with UI frameworks, installation, and cross-platform support. We've used `tk` and `wx` before and neither was easy
2. We adopted the server/web setup as this is fairly typical of modern UIs and it is fairly easy to find developers who can work with HTML/JS UIs
3. Vue.js was chosen because it could be embedded without the need for "building" the webpage with webpack, transpilers, vendor bundles, etc...as this eases development and installation

Not the best rationale, but it is what we got.

**Reply by ThibFrgsGmz** [reply]
Okay, thanks for the valuable feedback! :)
